modules/aws-auth/variables.tf

```diff
@@ -0,0 +1,114 @@
+# --------------------------------------------------------------------------------------------------
+# REQUIRED PARAMETERS
+# You must provide a value for each of these parameters.
+# --------------------------------------------------------------------------------------------------
+variable "vault_iam_role_id" {
+  description = "Vault IAM role ID to apply the policy to"
+}
+
+variable "consul_iam_role_arn" {
+  description = "ARN of the IAM role for Consul servers"
+}
+
+variable "nomad_server_iam_role_arn" {
+  description = "ARN of the IAM role for Nomad servers"
+}
+
+variable "nomad_client_iam_role_arn" {
+  description = "ARN of the IAM role for Nomad clients"
+}
+
+variable "vault_iam_role_arn" {
+  description = "ARN of the IAM role for Vault servers"
+}
+
+# --------------------------------------------------------------------------------------------------
+# OPTIONAL PARAMETERS
+# These parameters have reasonable defaults.
+# --------------------------------------------------------------------------------------------------
+variable "aws_auth_path" {
+  description = "Path to enable the AWS authentication method on"
+  default     = "aws/"
+}
+
+variable "iam_policy_name" {
+  description = "Name of the IAM policy to allow Vault servers to authenticate with AWS"
+  default     = "VaultAwsAuth"
+}
+
+variable "base_policies" {
+  description = "List of policies to assign to all tokens created via the AWS authentication method"
+  default     = []
+}
+
+variable "consul_role" {
+  description = "Name of the AWS authentication role for Consul servers"
+  default     = "consul"
+}
+
+variable "consul_policies" {
+  description = "Policies to attach to Consul servers role"
+  default     = []
+}
+
+variable "nomad_server_role" {
+  description = "Name of the AWS authentication role for Nomad servers"
+  default     = "nomad-server"
+}
+
+variable "nomad_server_policies" {
+  description = "Policies to attach to Nomad servers role"
+  default     = []
+}
+
+variable "nomad_client_role" {
+  description = "Name of the AWS authentication role for Nomad clients"
+  default     = "nomad-client"
+}
+
+variable "nomad_client_policies" {
+  description = "Policies to attach to Nomad clients role"
+  default     = []
+}
+
+variable "vault_role" {
+  description = "Name of the AWS authentication role for Vault servers"
+  default     = "vault"
+}
+
+variable "vault_policies" {
+  description = "Policies to attach to Vault servers role"
+  default     = []
+}
+
+variable "period_minutes" {
+  description = <<EOF
+The token should be renewed within the duration specified by this value.
+At each renewal, the token's TTL will be set to the value of this field.
+The maximum allowed lifetime of token issued using this role. Specified as a number of minutes.
+EOF
+
+  default = 4320
```

Comment:
Should a period be defined? Would the source token be required anymore upon the creation of the periodic token for the respective roles? for e.g. nomad server token (https://github.com/GovTechSG/terraform-modules/pull/63/files#diff-bd30aabf92f8b61fc984189b2ffd2c80R4)

Reply:
Unless I orphan them. I don't want to orphan them so that if we take an EC2 instance out of service all the children tokens should die too.

Except for the tokens that Nomad servers generate for jobs. Nomad servers will take care of that for us.